### ISSUE TITLE
Error compatibility embroider

### ISSUE THREAD

**Issue by villander** [ember-intl/ember-intl#858]
### Environment

- Ember Version: 3.9.0
- Ember CLI Version: 3.9.0
- Ember Intl Version: 3.5.2
- Embroider 3.4.0
- Browser(s): Chrome
- Node Version: 10

### Steps to Reproduce

`ember s`

```
jquery.js:3827 Uncaught TypeError: Cannot read property 'localeFactory' of undefined
    at Class.translationsFor (intl.js:196)
    at Class.addTranslations (intl.js:190)
    at eval (hydrate.js:33)
    at Array.forEach (<anonymous>)
    at eval (hydrate.js:31)
    at Class.init (intl.js:106)
    at Class.superWrapper [as init] (utils.js:365)
    at initialize (core_object.js:83)
    at Function.create (core_object.js:617)
    at FactoryManager.create (container.js:549)
translationsFor @ intl.js:196
addTranslations @ intl.js:190
(anonymous) @ hydrate.js:33
(anonymous) @ hydrate.js:31
init @ intl.js:106
superWrapper @ utils.js:365
initialize @ core_object.js:83
create @ core_object.js:617
create @ container.js:549
instantiateFactory @ container.js:359
lookup @ container.js:287
lookup @ container.js:131
lookup @ container_proxy.js:78
initialize @ initializer.js:17
(anonymous) @ index.js:132
Vertices.each @ dag-map.js:228
Vertices.walk @ dag-map.js:142
DAG.each @ dag-map.js:72
DAG.topsort @ dag-map.js:80
_runInitializer @ index.js:147
runInstanceInitializers @ index.js:130
_bootSync @ instance.js:108
didBecomeReady @ application.js:689
invoke @ backburner.js:335
flush @ backburner.js:226
flush @ backburner.js:423
_end @ backburner.js:957
end @ backburner.js:707
_run @ backburner.js:1010
_join @ backburner.js:984
join @ backburner.js:757
join @ index.js:165
(anonymous) @ index.js:260
mightThrow @ jquery.js:3534
process @ jquery.js:3602
setTimeout (async)
jQuery.readyException @ jquery.js:3826
(anonymous) @ jquery.js:3846
mightThrow @ jquery.js:3534
process @ jquery.js:3602
setTimeout (async)
(anonymous) @ jquery.js:3640
fire @ jquery.js:3268
fireWith @ jquery.js:3398
fire @ jquery.js:3406
fire @ jquery.js:3268
fireWith @ jquery.js:3398
process @ jquery.js:3622
setTimeout (async)
(anonymous) @ jquery.js:3640
fire @ jquery.js:3268
fireWith @ jquery.js:3398
fire @ jquery.js:3406
fire @ jquery.js:3268
fireWith @ jquery.js:3398
ready @ jquery.js:3878
completed @ jquery.js:3888
Show 5 more frames
```

this erros happens on this function

```js
/** @public **/
  translationsFor(localeName) {
    return this._adapter.localeFactory(Object(_private_normalize_locale__WEBPACK_IMPORTED_MODULE_4__["default"])(localeName));
  },
```

**Comment by jasonmit** [comment]
I haven't begun using embroider for anything, would appreciate any help in landing a fix.

**Comment by buschtoens** [comment]
This lookup must be failing:

https://github.com/ember-intl/ember-intl/blob/860b051103ee88cec84c4d7cc703ed4714f3a187/addon/services/intl.js#L105

I guess that [`addon/adapters/default.js`](https://github.com/ember-intl/ember-intl/blob/master/addon/adapters/default.js) is not ending up in the final build. It might be shaken out from the tree?

**Comment by jasonmit** [comment]
Is this still an issue?  Can anyone throw together a reproduction or include the steps.

**Comment by efx** [comment]
I was unable to reproduce this using a [newly created application.](https://github.com/efx/embroider-ember-intl). Though, the primary app I develop in does throw this error when built with embroider.
The below screenshot of the failing stack trace is from the private app:
<img width="1408" alt="Screen Shot 2019-06-06 at 14 32 08" src="https://user-images.githubusercontent.com/472721/59057251-f9795c00-8867-11e9-87b4-12853fa59dab.png">

**Comment by efx** [comment]
Upgrading to ember-intl@4.0.0-beta.10 fixes this error for me. embroider version:
```
		"@embroider/compat": "^0.3.5",
		"@embroider/core": "^0.3.5",
		"@embroider/webpack": "^0.3.5",
```
Should we close this by recommending using the latest version of ember-intl? That wouldn't be helpful for those needing to support < 4.0. I have no qualms updating our app but that is one user's opinion. 😄

**Comment by jasonmit** [comment]
@efx since 4.x is the latest stable I don't feel it's worth documenting.  If more folks open issues relating to it without upgrading first I'll update the docs.

**Comment by martony38** [comment]
I am still seeing this error with ember-intl@4.0.1 when using embroider. I was able to fix it by adding the following package rule to `ember-cli-build.js`:
```js
packageRules: [
  {
    package: 'ember-intl',
    semverRange: '^4.0.1',
    addonModules: {
      'services/intl.js': { 
        dependsOnModules: ['../adapters/default.js']
      }
    }
  },
  ...
]
```

**Comment by NullVoxPopuli** [comment]
@martony38 I just ran in to this issue with ember-intl and your packageRule fixed it for me :)
do you have entries for `@ember-data/model/-private` and `ember-destroyable-polyfill/-internal/patch-meta`? 

Related:
 - https://github.com/ember-polyfills/ember-destroyable-polyfill/issues/12

**Comment by martony38** [comment]
@NullVoxPopuli nope :man_shrugging:

**Comment by NullVoxPopuli** [comment]
looks like we need a different config for ember-intl v5

the error I'm getting with v5.2.0:
```
Error while processing route: index intl_messageformat_parser__WEBPACK_IMPORTED_MODULE_0__.default is undefined parseString@webpack:///../node_modules/ember-intl/-private/utils/parse.js?:11:3
compile@webpack:///../node_modules/ember-intl/-private/store/translation.js?:25:72
addTranslations@webpack:///../node_modules/ember-intl/-private/store/translation.js?:38:37
push@webpack:///../node_modules/ember-intl/-private/store/container.js?:36:48
addTranslations@webpack:///../node_modules/ember-intl/services/intl.js?:186:32
hydrate/<@webpack:///../node_modules/ember-intl/-private/utils/hydrate.js?:12:13
hydrate@webpack:///../node_modules/ember-intl/-private/utils/hydrate.js?:11:57
init@webpack:///../node_modules/ember-intl/services/intl.js?:97:75
superWrapper@https://localhost:4201/assets/vendor.js:30902:22
initialize@https://localhost:4201/assets/vendor.js:29228:9
create@https://localhost:4201/assets/vendor.js:29777:19
create@https://localhost:4201/assets/vendor.js:1230:33
instantiateFactory@https://localhost:4201/assets/vendor.js:1023:71
lookup@https://localhost:4201/assets/vendor.js:951:12
lookup@https://localhost:4201/assets/vendor.js:792:14
lookup@https://localhost:4201/assets/vendor.js:27489:33
getInjection@https://localhost:4201/assets/vendor.js:17250:20
get/<@https://localhost:4201/assets/vendor.js:15520:32
untrack@https://localhost:4201/assets/vendor.js:55943:7
get@https://localhost:4201/assets/vendor.js:15519:34
CPGETTER_FUNCTION@https://localhost:4201/assets/vendor.js:14509:25
setLocale@webpack:///./services/locale.ts?:31:5
beforeModel@webpack:///./routes/application.ts?:35:23
runBeforeModelHook@https://localhost:4201/assets/vendor.js:62359:31
resolve/<@https://localhost:4201/assets/vendor.js:62268:128
invokeCallback@https://localhost:4201/assets/vendor.js:64334:17
publish@https://localhost:4201/assets/vendor.js:64317:23
@https://localhost:4201/assets/vendor.js:58757:61
invoke@https://localhost:4201/assets/vendor.js:56960:16
flush@https://localhost:4201/assets/vendor.js:56851:19
flush@https://localhost:4201/assets/vendor.js:57048:21
_end@https://localhost:4201/assets/vendor.js:57582:34
Backburner/this._boundAutorunEnd@https://localhost:4201/assets/vendor.js:57251:14
router.js:1212
TypeError: intl_messageformat_parser__WEBPACK_IMPORTED_MODULE_0__.default is undefined
```

**Comment by jasonmit** [comment]
Thanks @NullVoxPopuli for fixing the above issue ❤ Released in v5.2.2
